src/config/mod.rs

```diff
@@ -218,6 +252,24 @@ fn parse_config(path: &std::path::Path) -> Result<(Vec<IpAddr>, Vec<CtrEntry>),
                 bind_addrs.push(local_ip);
             }
 
+            // If network parts contain more than one col then
+            // we have custom dns server also defined at network level
+            // lets process that.
+            if network_parts.len() > 1 {
```

Comment:
does it make sense to invert this logic, and then put lines 221-222 in the conditional?  And then move the iteration outside the conditional?